server/service/src/document/patient/patient_document_update.rs

```diff
@@ -159,25 +164,22 @@ mod test {
             },
             allergies: None,
         };
+
         service
-            .update_document(
+            .update_patient(
                 &ctx,
-                "store_a",
-                RawDocument {
-                    name: "test/doc".to_string(),
-                    parents: vec![],
-                    author: "some_user".to_string(),
-                    timestamp: Utc::now(),
-                    r#type: PATIENT_TYPE.to_string(),
+                &service_provider,
+                "store_a".to_string(),
+                "user",
+                UpdatePatient {
                     data: serde_json::to_value(patient.clone()).unwrap(),
-                    // TODO add and use patient id schema
-                    schema_id: None,
+                    schema_id: schema.id.clone(),
```

Comment:
interesting that the patient identifier is only within the JSON, wouldn't that be something that should be lifted higher? and passed in here when updating?

Reply:
Think its good to have the patient id as part of the patient document. So I would have to validate that both ids match... I also don't want to split the document in default fields and custom fields so that I have to compose it at a later stage. This endpoint is basically still just an endpoint to update a document, the only difference is that I can only mutate patient document of a defined shape.